### PR TITLE
fix: inconsistent behavior between `signature_namespace` and `signature_types`

### LINKED ISSUE #3681: bug: inconsistent behavior between `signature_namespace` and `signature_types`
It's currently difficult to use the simpler `signature_types` syntax in plugins due to the potential of there being an exception raised during startup. 

We should consider removing this entire section to keep the behavior consistent between `signature_namespace` and `signature_types`.

https://github.com/litestar-org/litestar/blob/238d26bab30afa1f852ef0d88830077aab77ba35/tests/unit/test_utils/test_signature.py#L159-L168

---
> [!NOTE]  
> While we are open for sponsoring on [GitHub Sponsors](https://github.com/sponsors/litestar-org/) and 
> [OpenCollective](https://opencollective.com/litestar), we also utilize [Polar.sh](https://polar.sh/) to engage in pledge-based sponsorship.
>
> Check out all issues funded or available for funding [on our Polar.sh dashboard](https://polar.sh/litestar-org)
> * If you would like to see an issue prioritized, make a pledge towards it!
> * We receive the pledge once the issue is completed & verified
> * This, along with engagement in the community, helps us know which features are a priority to our users.

<a href="https://polar.sh/litestar-org/litestar/issues/3681">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/3681/pledge.svg?darkmode=1">
  Fund with Polar
</picture>
</a>

### PR DESCRIPTION
**Pull request by cofin**
Fixes #3681

## Description

-

Closes #3681

### CODE REVIEW

**Comment by provinzkraut** [comment]
So, this wouldn't warn by default? 
Is there a reason you went with this logger approach instead of the standard warning stuff we usually do?

File: litestar/utils/signature.py [hunk 1]
```diff
@@ -253,16 +254,46 @@ def add_types_to_signature_namespace(
...
-            raise ImproperlyConfiguredException(f"Type '{name}' is already defined in the signature namespace")
-        signature_namespace[name] = typ
+    return merge_signature_namespaces(
+        signature_namespace=signature_namespace,
+        additional_signature_namespace={signature_type.__name__: signature_type for signature_type in signature_types},
+        logger=logger,
+    )
+
+
+def merge_signature_namespaces(
```
Comment (provinzkraut) [inline litestar/utils/signature.py:272]: Should use this in `BaseRouteHandler.resolve_signature_namespace` as well for consistency :)

**Comment by cofin** [comment]
> So, this wouldn't warn by default? Is there a reason you went with this logger approach instead of the standard warning stuff we usually do?

Yes.  I really don't think this should be a warning at all.  It should just be something the user is aware of.

For instance, if a plugin sets something automatically and then the user sets their own.  I think we should just notify the user that there could be a problem.  I wouldn't consider this an actual warning.

**Comment by provinzkraut** [comment]
>  I think we should just notify the user that there could be a problem. I wouldn't consider this an actual warning.

I'm with you there, but we aren't doing that, because we don't pass a logger by default, right? 
I feel like I'm missing something here :grimacing:

File: tests/unit/test_utils/test_signature.py [hunk 2]
```diff
@@ -156,16 +159,29 @@ def test_add_types_to_signature_namespace() -> None:
     assert ns == {"int": int, "str": str}
 
 
-def test_add_types_to_signature_namespace_with_existing_types() -> None:
+def test_add_types_to_signature_namespace_no_warn() -> None:
     """Test add_types_to_signature_namespace with existing types."""
-    ns = add_types_to_signature_namespace([str], {"int": int})
-    assert ns == {"int": int, "str": str}
+    if os.environ.get("LITESTAR_WARN_SIGNATURE_NAMESPACE_OVERRIDE", None):
+        del os.environ["LITESTAR_WARN_SIGNATURE_NAMESPACE_OVERRIDE"]
```
Comment (provinzkraut) [inline tests/unit/test_utils/test_signature.py:165]:
```suggestion
    monkeypatch.delenv("LITESTAR_WARN_SIGNATURE_NAMESPACE_OVERRIDE", raising=False)
```

So it's resets things properly. Easier to reason about IMO

File: tests/unit/test_utils/test_signature.py [hunk 3]
```diff
@@ -156,16 +159,29 @@ def test_add_types_to_signature_namespace() -> None:
...
+        add_types_to_signature_namespace([int], {"int": int})
 
 
-def test_add_types_to_signature_namespace_with_existing_types_raises() -> None:
+def test_add_types_to_signature_namespace_with_existing_types_warn() -> None:
     """Test add_types_to_signature_namespace with existing types raises."""
-    with pytest.raises(ImproperlyConfiguredException):
-        add_types_to_signature_namespace([int], {"int": int})
+    if os.environ.get("LITESTAR_WARN_SIGNATURE_NAMESPACE_OVERRIDE", None):
+        del os.environ["LITESTAR_WARN_SIGNATURE_NAMESPACE_OVERRIDE"]
```
Comment (provinzkraut) [inline tests/unit/test_utils/test_signature.py:174]: Same here

File: tests/unit/test_utils/test_signature.py [hunk 4]
```diff
@@ -156,16 +159,29 @@ def test_add_types_to_signature_namespace() -> None:
...
+    with pytest.warns(LitestarWarning):
+        add_types_to_signature_namespace([int], {"int": str})
+
+
+def test_add_types_to_signature_namespace_warn_disabled() -> None:
+    """Test add_types_to_signature_namespace with existing types."""
+    os.environ["LITESTAR_WARN_SIGNATURE_NAMESPACE_OVERRIDE"] = "0"
```
Comment (provinzkraut) [inline tests/unit/test_utils/test_signature.py:181]:
```suggestion
    monkeypatch.setenv("LITESTAR_WARN_SIGNATURE_NAMESPACE_OVERRIDE")
```

**Review by provinzkraut — APPROVED** [review]
Great! Just one more comment regarding the env test setup

**Pull request merged**